### PR TITLE
Adds a plugin for heptio-e2e tests

### PR DESCRIPTION
**Pull request by chuckha**
Closes #139

Signed-off-by: Chuck Ha <chuck@heptio.com>

### CODE REVIEW

**Comment by timothysc** [comment]
@chuckha you'll need to configure the sonobuoy-plugins-deployer project for jenkins.  I copied the project, but we'll need to enable.  

That's a good work item to enqueue.

**Comment by chuckha** [comment]
wow. total face palm. forgot to add the actual important file to the commit. thanks CI!

**Comment by timothysc** [comment]
Deployer for sonobuoy-plugins works properly this should not be unblocked.

File: pkg/config/loader_test.go [hunk 1]
```diff
@@ -108,6 +108,7 @@ func TestLoadAllPlugins(t *testing.T) {
 		PluginSelections: []plugin.Selection{
 			plugin.Selection{Name: "systemd_logs"},
 			plugin.Selection{Name: "e2e"},
+			plugin.Selection{Name: "heptio-check"},
```
Comment (timothysc) [inline pkg/config/loader_test.go]: heptio-e2e - check is on the UI side.

File: plugins.d/heptio-e2e.tmpl [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    sonobuoy-result-type: heptio-check
+  labels:
+    component: sonobuoy
+    sonobuoy-run: '{{.SessionID}}'
+    tier: analysis
+  name: heptio-check
+  namespace: '{{.Namespace}}'
+spec:
+  containers:
+    - image: 'gcr.io/heptio-images/heptio-e2e:latest'
```
Comment (timothysc) [inline plugins.d/heptio-e2e.tmpl]: :master for now, and we'll tag latest

**Comment by timothysc** [comment]
@k8k - once this is merged you should be unblocked on the scanner side.

File: pkg/config/loader_test.go (hunk 1 above)
Comment (chuckha) [reply pkg/config/loader_test.go]: yeah, fair, I'll rename it.

File: plugins.d/heptio-e2e.tmpl [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.nodeName
+        - name: RESULTS_DIR
+          value: /tmp/results
+        - name: MASTER_URL
+          value: '{{.MasterAddress}}'
+        - name: RESULT_TYPE
+          value: heptio-e2e
+      image: 'gcr.io/heptio-images/sonobuoy:latest'
```
Comment (timothysc) [inline plugins.d/heptio-e2e.tmpl]: This should also be :master if from the master repo.  

Only scanner would leverage :latest.

**Pull request merged**